rust-runtime/aws-smithy-runtime/src/client/identity/cache/lazy.rs

```diff
@@ -0,0 +1,671 @@
+    pub fn time_source(mut self, time_source: impl TimeSource + 'static) -> Self {
+        self.set_time_source(time_source.into_shared());
+        self
+    }
+    /// Set the time source for this cache.
+    pub fn set_time_source(&mut self, time_source: SharedTimeSource) -> &mut Self {
+        self.time_source = Some(time_source.into_shared());
+        self
+    }
+
+    /// Set the async sleep implementation for this cache.
+    pub fn sleep_impl(mut self, sleep_impl: impl AsyncSleep + 'static) -> Self {
+        self.set_sleep_impl(sleep_impl.into_shared());
+        self
+    }
+    /// Set the async sleep implementation for this cache.
+    pub fn set_sleep_impl(&mut self, sleep_impl: SharedAsyncSleep) -> &mut Self {
+        self.sleep_impl = Some(sleep_impl);
+        self
+    }
+
+    /// Timeout for identity resolution.
+    ///
+    /// Defaults to 5 seconds.
+    pub fn load_timeout(mut self, timeout: Duration) -> Self {
+        self.set_load_timeout(Some(timeout));
+        self
+    }
+
+    /// Timeout for identity resolution.
+    ///
+    /// Defaults to 5 seconds.
+    pub fn set_load_timeout(&mut self, timeout: Option<Duration>) -> &mut Self {
+        self.load_timeout = timeout;
+        self
+    }
+
+    /// Amount of time before the actual identity expiration time where the identity is considered expired.
+    ///
+    /// For example, if the identity are expiring in 15 minutes, and the buffer time is 10 seconds,
+    /// then any requests made after 14 minutes and 50 seconds will load a new identity.
+    ///
+    /// Defaults to 10 seconds.
+    pub fn buffer_time(mut self, buffer_time: Duration) -> Self {
+        self.set_buffer_time(Some(buffer_time));
+        self
+    }
+
+    /// Amount of time before the actual identity expiration time where the identity is considered expired.
+    ///
+    /// For example, if the identity are expiring in 15 minutes, and the buffer time is 10 seconds,
+    /// then any requests made after 14 minutes and 50 seconds will load a new identity.
+    ///
+    /// Defaults to 10 seconds.
+    pub fn set_buffer_time(&mut self, buffer_time: Option<Duration>) -> &mut Self {
+        self.buffer_time = buffer_time;
+        self
+    }
+
+    /// A random percentage by which buffer time is jittered for randomization.
+    ///
+    /// For example, if the identity is expiring in 15 minutes, the buffer time is 10 seconds,
+    /// and buffer time jitter fraction is 0.2, then buffer time is adjusted to 8 seconds.
+    /// Therefore, any requests made after 14 minutes and 52 seconds will load a new identity.
+    ///
+    /// Defaults to a randomly generated value between 0.0 and 1.0. This setter is for testing only.
+    #[cfg(feature = "test-util")]
+    pub fn buffer_time_jitter_fraction(mut self, buffer_time_jitter_fraction: fn() -> f64) -> Self {
+        self.set_buffer_time_jitter_fraction(Some(buffer_time_jitter_fraction));
+        self
+    }
```

Comment:
does this need to be pub, even feature gated?

Reply:
It doesn't look like this is even used in tests anymore. I'm going to make it private and `cfg(test)` in case its needed in the future.